.github/workflows/CD.yml

```diff
@@ -0,0 +1,116 @@
+name: CD
+
+on:
+  push:
+    tags:        
+      - '**' 
+
+jobs:
+  validate-tag:
+    name: Validate tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+      
+      - name: Check no jumps in tags
+        run: |
+          # If a previous tag is X.Y, valid tags are only X.(Y+1) or (X+1).0
+          if [[ $(git tag -l | wc -l) != 1 ]]; then # If there is only one tag, no need to check
+            prev=($(git tag -l --sort=-v:refname | head -n 2 | tail -n 1 | tr "." " "))
+            val1="${prev[0]}.$((${prev[1]}+1))" # X.(Y+1)
+            val2="$((${prev[0]}+1)).0" # (X+1).0
+            if [[ "${{github.ref_name}}" != "$val1" ]] && \
+            [[ "${{github.ref_name}}" != "$val2" ]]; then
+                echo "Invalid tag \"${{github.ref_name}}\" when previous tag is 
+                \"$(sed 's/ /./' <<< ${prev[@]})\"."
+                exit 1
+            fi
+          fi
+  
+  get-package-name:
+    name: Get package name
+    runs-on: ubuntu-latest
+    needs: validate-tag
+    outputs:
+      package-name: ${{ steps.get-package-name.outputs.package-name }}
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+      
+      - name: Get name
+        id: get-package-name
+        run: |
+          echo "package-name=$(yq -oy '.project.name' pyproject.toml)" >> $GITHUB_OUTPUT
+  
+  release-conda-package:
+    name: Build with conda and release
+    runs-on: ubuntu-latest
+    needs: [get-package-name, validate-tag]
+    env:
+      BUILD_FOLDER: ${{github.workspace}}/build
+      PACKAGE_PATH: ${{github.workspace}}/build/noarch/*.tar.bz2
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+            fetch-tags: true
+            fetch-depth: 0 
+
+      - name: Setup conda build environment
+        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3.0.4
+        with:
+          miniconda-version: "latest"
+          python-version: ${{ vars.PY_VERSION }} 
+          environment-file: .conda/env_build.yml
+          auto-activate-base: false
+          auto-update-conda: false
+          show-channel-urls: true
+      
+      - name: Build conda package
+        shell: bash -el {0}
+        run: |
+          conda build . --no-anaconda-upload --output-folder=${{env.BUILD_FOLDER}} -c conda-forge -c accessnri -c coecms
```

Comment:
Do we need the `coecms` channel still? I heard it is shutting down soonish?

Reply:
Yes, we still need it for `mule`. 
We should try to find a solution, but for now we have to use it.

Reply:
I think we want the channel preference to be `accessnri` > `conda-forge`.

I'm thinking it might be better to specify this once in a single location?

Can we make (and link) an issue to get rid of the `coecms` dependency?

Reply:
Order has been changed. 

For the specification in a single location, I agree it could be better placed (so we don't have to repeat it in multiple workflows), but where would you specify it? 

These channels are likely the same for all the packages we deploy (even if `coecms` could change and would not be needed for the packages that don't use `mule`, it should not be a big issue adding it to packages that don't need it).
I think it could be suitable as an Organization variable.
Something like:
- variable_name: CONDA_CHANNELS
- value: `-c accessnri -c conda-forge -c coecms` (`coecms` would probably be removed when possible)

And we could use it in any repo, for example, as:
```
conda build . --no-anaconda-upload --output-folder=${{env.BUILD_FOLDER}} ${{vars.CONDA_CHANNELS}}
```

Reply:
I think it sounds like a good idea. I'd like @CodeGat's opinion and I think we should make an issue to discuss. There isn't an obvious code repository to make an issue about it, so I'd suggest making one in dev-docs and let some of the other conda packaging groups know about it.

I don't think this should hold up this PR.

Reply:
_the disembodied voice of Tommy comes to you, as if in a dream..._
Myself and Jo were considering having a similar meta-repo as we do for model deployment and configuration stuff, but for conda packaging and deployment. Since we seem to do be doing it a lot.

Reply:
Created https://github.com/ACCESS-NRI/dev-docs/issues/31